src/SmoFacade/Database.cs

```diff
@@ -45,21 +45,30 @@ public void RestoreWithRecovery()
     /// </summary>
     public void Drop()
     {
-      var policy = Policy
-        .Handle<FailedOperationException>()
-        .Or<TimeoutException>()
+      var policyPrep = Policy
+        .Handle<Exception>()
         .WaitAndRetry(4, retryAttempt => TimeSpan.FromMilliseconds(Math.Pow(10, retryAttempt)));
 
       // ensure database is not in AvailabilityGroup, WaitAndRetry loop for each instance to sync
-      policy.Execute(() => {
-        if(string.IsNullOrEmpty(_database.AvailabilityGroupName))
-          return;
-        if(_database.AvailabilityDatabaseSynchronizationState >
-           AvailabilityDatabaseSynchronizationState.NotSynchronizing)
-          throw new
-            TimeoutException($"Cannot kill the database {Name} until it has been removed from the AvailabilityGroup.");
+      policyPrep.Execute(() => {
+        if (!string.IsNullOrEmpty(_database.AvailabilityGroupName))
+          throw new Exception($"Cannot kill the database {Name} until it has been removed from the AvailabilityGroup");

```

Comment:
Can we throw something more specific than exception? Or do we truly want to retry on every type of exception?

Reply:
What other exception types are possible here?  All its doing is retrieving an Enum property.
I'm not sure what benefit a different exception type would make here

Reply:
That's the point, I don't care to dig into the database object and likely into SMO to find out what other exceptions we may be retrying. Using a uniquely typed exception for what we want to retry on makes more sense, unless we're saying we want to retry on everything even not knowing what everything is.

Reply:
I want this to retry on everything, yes.
If SMO fails to retrieve one of these properties, throwing anything other than `PropertyCannotBeRetrievedException`, it should retry